File: packages/adapters/pkg/utils/validation_test.go
package utils

import (
	"strings"
	"testing"
)

func TestValidation(t *testing.T) {
	t.Run("IsValidEmail", func(t *testing.T) {
		assert := func(email string, expected bool) {
			if got := IsValidEmail(email); got != expected {
				t.Errorf("IsValidEmail(%q) = %v, want %v", email, got, expected)
			}
		}
		assert("test@example.com", true)
		assert("test.name@example.com", true)
		assert("USER.Name+tag@example.com", true)
		assert("test@sub.example.technology", true)
		assert("", false)
		assert(strings.Repeat("a", MaxEmailLength+1), false)
		assert("invalid-email", false)
		assert("@example.com", false)
		assert("test@", false)
		assert(".user@example.com", false)
		assert("user.@example.com", false)
		assert("a..b@example.com", false)
		assert("user@-example.com", false)
		assert("user@example..com", false)
		assert(strings.Repeat("a", 64)+"@example.com", true)
		assert(strings.Repeat("a", 65)+"@example.com", false)
	})
}

File: packages/core/ts/client-core/src/utils/validation.test.ts
import { describe, expect, it } from 'bun:test';

import { isValidEmail, isValidUrl } from './validation';

describe('utils/validation', () => {
  it('accepts syntactically valid email addresses', () => {
    expect(isValidEmail('user@example.com')).toBe(true);
    expect(isValidEmail('first.last+tag@sub.example.co')).toBe(true);
  });

  it('rejects invalid email addresses', () => {
    expect(isValidEmail('')).toBe(false);
    expect(isValidEmail('missing-at.example.com')).toBe(false);
    expect(isValidEmail('user@')).toBe(false);
  });

  it('accepts only http and https urls', () => {
    expect(isValidUrl('https://taskforce.ai')).toBe(true);
    expect(isValidUrl('http://localhost:3000/path?x=1')).toBe(true);
    expect(isValidUrl('ftp://taskforce.ai')).toBe(false);
    expect(isValidUrl('javascript:alert(1)')).toBe(false);
    expect(isValidUrl('not a url')).toBe(false);
  });
});
